src/autopas/containers/octree/OctreeDirection.h

```diff
@@ -0,0 +1,532 @@
+/**
+ * @file OctreeInnerNode.h
+ *
+ * @author Johannes Spies
+ * @date 18.05.2021
+ */
+
+#pragma once
+
+namespace autopas {
+
+/**
+ * This enum can be used to index the faces of a cube including an "invalid" face.
+ */
+enum Face {
+  O = 0,  // omega/unknown
+  L = 1,
+  R = 2,
+  D = 3,
+  U = 4,
+  B = 5,
+  F = 6,
+};
+
+/**
+ * A datatype that is wide enough to hold faces, edges or vertices.
+ */
+using Any = int unsigned;
+
+/**
+ * Create a bitfield for an edge given by the template parameters.
+ *
+ * @tparam f1
+ * @tparam f2
+ * @return
+ */
+template <Face f1, Face f2>
+static constexpr Any buildEdge() {
+  static_assert(f1 != f2, "Faces must be different");
+  return (f1 << 3) | f2;
+}
+
+/**
+ * Create a bitfield for a vertex given by the template parameters.
+ *
+ * @tparam f1
+ * @tparam f2
+ * @tparam f3
+ * @return
+ */
+template <Face f1, Face f2, Face f3>
+static constexpr Any buildVertex() {
+  static_assert((f1 != f2) && (f2 != f3), "Faces must be different");
+  return (f1 << 6) | (f2 << 3) | f3;
+}
+
+/**
+ * This enum can be used to index all edges of a cube including an "invalid" edge.
+ */
+enum Edge {
+  OO = 0,  // omega/unknown
+  LD = buildEdge<L, D>(),
+  LU = buildEdge<L, U>(),
+  LB = buildEdge<L, B>(),
+  LF = buildEdge<L, F>(),
+  RD = buildEdge<R, D>(),
+  RU = buildEdge<R, U>(),
+  RB = buildEdge<R, B>(),
+  RF = buildEdge<R, F>(),
+  DB = buildEdge<D, B>(),
+  DF = buildEdge<D, F>(),
+  UB = buildEdge<U, B>(),
+  UF = buildEdge<U, F>(),
+};
+
+/**
+ * This enum can be used to index all vertices of a cube including an "invalid" vertex.
+ */
+enum Vertex {
+  OOO = 0,  // omega/unknown
+  LDB = buildVertex<L, D, B>(),
+  LDF = buildVertex<L, D, F>(),
+  LUB = buildVertex<L, U, B>(),
+  LUF = buildVertex<L, U, F>(),
+  RDB = buildVertex<R, D, B>(),
+  RDF = buildVertex<R, D, F>(),
+  RUB = buildVertex<R, U, B>(),
+  RUF = buildVertex<R, U, F>(),
+};
+
+/**
+ * Get all available faces for a cube.
+ *
+ * @return A pointer to a static table. The last element is the "invalid" face O.
+ */
+inline Face *getFaces() {
+  static Face table[] = {L, R, D, U, B, F, O};
```

Comment:
If you wrap all of this in a namespace the probably better alternative would be to make these tables directly `constexpr static` variables and get rid of these getters.